lib/annotations/example.js

```diff
@@ -61,28 +61,43 @@ module.exports = (env) => {
             exampleItem.rendered = undefined;
             /* istanbul ignore else */
             if (!sass) {
+              if (
+                env.herman?.sass?.implementation &&
+                typeof env.herman.sass.implementation !== 'string'
+              ) {
+                sass = env.herman.sass.implementation;
+              } else {
+                const sassPkg = env.herman?.sass?.implementation ?? 'sass';
+                try {
+                  // eslint-disable-next-line global-require
+                  sass = require(sassPkg);
+                } catch (err) /* istanbul ignore next */ {
+                  env.logger.warn(
+                    `Cannot find Dart Sass (\`${sassPkg}\`) dependency, ` +
+                      'which is required when using the `@example scss` annotation.',
+                  );
+                  env.logger.warn(err);
+                  return;
+                }
+              }
+            }
+            /* istanbul ignore else */
+            if (sass && !sassCompiler) {
               try {
-                // eslint-disable-next-line global-require
-                sass = require('sass');
+                sassCompiler = sass.initAsyncCompiler();
```

Comment:
Option- I'm thinking it would be simpler and avoid a lot of subsequent `.then(compiler=>)` code if we could have `sassCompiler` be the compiler, and not the promise.

Reply:
Hm... 🤔 I'm not excited about refactoring everything here to use async/await, and I'm not sure I see a clean way using the current Promise structure to guard against race conditions using your approach. I'm open to it if you have a proposal I'm missing, but otherwise since it's only used in two places total I'm also okay leaving it as-is.

Reply:
That makes sense to me!